### PR TITLE
Nice errors and tasklet arguments for the Python frontend

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

**Review by alexnick83 — CHANGES_REQUESTED** [review]
The main changes introduced by this PR (capability for side-effect explicit dataflow tasklets and changes in how an explicit dataflow tasklet's language is interpreted/parsed) need examples/tests. I would like to see a few because I have difficulty understanding the features this PR adds.

File: dace/frontend/python/interface.py [hunk 1]
```diff
@@ -293,10 +293,11 @@ class tasklet(metaclass=TaskletMetaclass):
     The DaCe framework cannot analyze these tasklets for optimization. 
     """
 
-    def __init__(self, language: Union[str, dtypes.Language] = dtypes.Language.Python):
+    def __init__(self, language: Union[str, dtypes.Language] = dtypes.Language.Python, side_effects: bool = False):
         if isinstance(language, str):
             language = dtypes.Language[language]
         self.language = language
+        self.side_effects = side_effects
```
[the remark this answers is not in the document]
Comment (tbennun) [reply dace/frontend/python/interface.py:300]: Any `dace.tasklet` with a printout and without an output will be removed by dead dataflow elimination. It is strange to add an output dependency there manually. That is the use case here (external function calls).

File: dace/frontend/python/memlet_parser.py [hunk 2]
```diff
@@ -285,7 +285,11 @@ def ParseMemlet(visitor,
         if len(node.value.args) >= 2:
             write_conflict_resolution = node.value.args[1]
 
-    subset, new_axes, arrdims = parse_memlet_subset(array, node, das, parsed_slice)
+    try:
```
[the remark this answers is not in the document]
Comment (tbennun) [reply dace/frontend/python/memlet_parser.py:288]: It contains no information at all (just IndexError) and no line provenance, so the dace-specific syntax error will even give you the line of code it occurred in.

**Comment by tbennun** [comment]
@alexnick83 what should we do with this PR? I answered the comments now.

**Review by alexnick83 — APPROVED** [review]

**Pull request merged**